crates/aggchain-proof-builder/src/config.rs

```diff
@@ -11,6 +12,10 @@ pub struct AggchainProofBuilderConfig {
     /// ID of the network for which the proof is generated (rollup id).
     pub network_id: u32,
 
+    /// Hash of the aggregated execution proof verification key.
+    #[serde(default = "default_aggregated_proof_vkey_hash")]
+    pub aggregated_proof_vkey_hash: B256,
```

Comment:
A better place for this could be in the `proposer-service`, this service is responsible for requesting and returning a valid aggspan-proof, thus checking the vKey makes more sense in this component.

Reply:
Make sense. If the vkey is invalid, there is not point making the `aggchain-proof-builder` request to generate a proof.
Then we would need to move this expected vkey config parameter and `check_aggspan_proof_vkey` to the proposer service, and `agghcain-proof-builder` would just proceed with the agghain proof generation.